src/test/java/net/sf/jabref/importer/fileformat/RepecNepImporterTest.java

```diff
@@ -0,0 +1,122 @@
+package net.sf.jabref.importer.fileformat;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import net.sf.jabref.Globals;
+import net.sf.jabref.JabRefPreferences;
+import net.sf.jabref.bibtex.BibtexEntryAssert;
+import net.sf.jabref.importer.OutputPrinterToNull;
+import net.sf.jabref.model.entry.BibEntry;
+
+public class RepecNepImporterTest {
+
+    @Before
+    public void setUp() throws Exception {
+        if (Globals.prefs == null) {
+            Globals.prefs = JabRefPreferences.getInstance();
+        }
+    }
+
+    @Test
+    public final void testIsRecognizedFormat() throws IOException {
+        RepecNepImporter testImporter = new RepecNepImporter();
+
+        //test with non recognized format files
+        List<String> notAccepted = Arrays.asList("RepecNep1.xml", "CopacImporterTest1.txt", "RisImporterTest1.ris",
+                "CopacImporterTest2.txt", "IEEEImport1.txt");
+        for (String s : notAccepted) {
+            try (InputStream stream = RepecNepImporter.class.getResourceAsStream(s)) {
+                Assert.assertFalse(testImporter.isRecognizedFormat(stream));
+            }
+        }
+
+        //test with recognized format files
+        List<String> accepted = Arrays.asList("RepecNepImporterTest1.txt", "RepecNepImporterTest2.txt",
+                "RepecNepImporterTest3.txt");
+        for (String s : accepted) {
+            try (InputStream stream = RepecNepImporter.class.getResourceAsStream(s)) {
+                Assert.assertTrue(testImporter.isRecognizedFormat(stream));
+            }
+        }
+    }
+
+    @Test
+    public final void testImportEntries() throws IOException {
+        RepecNepImporter testImporter = new RepecNepImporter();
+
+        List<BibEntry> entries = testImporter.importEntries(
+                RepecNepImporter.class.getResourceAsStream("RepecNepImporterTest1.txt"), new OutputPrinterToNull());
+        Assert.assertEquals(2, entries.size());
+
+        entries = testImporter.importEntries(RepecNepImporter.class.getResourceAsStream("RepecNepImporterTest3.txt"),
+                new OutputPrinterToNull());
+        Assert.assertEquals(17, entries.size());
+
+        entries = testImporter.importEntries(RepecNepImporter.class.getResourceAsStream("RepecNepImporterTest4.txt"),
+                new OutputPrinterToNull());
+        Assert.assertEquals(29, entries.size());
+
+        try {
```

Comment:
Have a look at the [expected exception rule](http://stackoverflow.com/questions/156503/how-do-you-assert-that-a-certain-exception-is-thrown-in-junit-4-tests)